composer.json

```diff
@@ -3,7 +3,7 @@
     "type": "typo3-flow-package",
     "description": "Create link to static resources with cache-breaking segment based on md5 of the file",
     "require": {
-        "neos/flow": "^4.0.0"
+        "neos/flow": "^5.0.0"
```

Comment:
Are there any breaking changes that would prevent it from working with 4.x, just curious?

Reply:
to be honest, I don't know and didn't check - if you could check that, I'd be happy to also cover neos 4.x with the same package version

Reply:
Just by skimming through the code I don't see anything breaking.

Reply:
Since there are no code changes, there is no update needed for the old neos

Reply:
Yep, but if we'll have new features in the future, they'd no longer be available to those still using Flow 4.x.
I'm not saying we should support Flow 4.x forever, but if it doesn't cost us any extra effort, why explicitly deny those who have not yet upgraded to use the updated version of the package?

Reply:
feel free to adjust the composer.json as you wish. It's your code ;)

Reply:
It's _our_ code ;)

Reply:
What about having
- v3.x compatible with Flow 5.x
- v2.x compatible with Flow 4.x

For now, I don't expect a big number of new features to be added - at least not from my side due to lack of time - and tbh I think the package is pretty feature-complete for now (but of course, as soon as one has an idea, let's talk about that).

Having the two branches IMO has the following benefits:
- even if the code is 1:1 or almost 1:1 in the two branches right now, composer automagically pulls the right version for the used Neos version
- if we have to make small changes (e.g. bugfixes) that are only relevant for the one or the other Neos branch - well, we can just do it in the relevant branch of this package
- if we add new features, they will target master branch and end up in v3.x "by default"
- if the feature shall also be available for older Neos branches (as in "Neos versions still supported by Neos *and* supported by us), we should be easily able to backport it to the 2.x branch if wanted

But maybe I'm missing an important point and we can change that.

Reply:
@mrimann that's of course the most mature solution, but that requires more hassle. I'd support Flow 4.x will soon go out of support anyways.